Library/Homebrew/extend/os/linux/requirements/java_requirement.rb

```diff
@@ -0,0 +1,24 @@
+require "language/java"
+
+class JavaRequirement < Requirement
+  fatal true
+  default_formula "jdk"
+  download "http://www.oracle.com/technetwork/java/javase/downloads/index.html"
+
+  satisfy build_env: false do
+    jdk = Formula["jdk"]
+    @java_home = jdk.opt_prefix if jdk && jdk.installed?
+    next quiet_system "java", "-version"
+  end
+
+  env do
+    next unless @java_home
+    java_home = Pathname.new(@java_home)
+    ENV["JAVA_HOME"] = java_home
+    ENV.prepend_path "PATH", java_home/"bin"
+    if (java_home/"include").exist? # Oracle JVM
```

Comment:
I'm not sure if this check is still necessary. I don't think we still need to worry about GCJ after all these years and the widespread availability of OpenJDK.

Thoughts?